SEED-platform/seed
# Fix media files and docker

#### Any background context you want to provide?
User uploaded content was not being saved in the media directory. If one tries to set the media root to something different, it will not be respected (and will be defaulted to the STATIC_ROOT).

#### What's this PR do?

- FIxes the MEDIA_ROOT when using local filestorage. Now the media (user uploaded content) and the static files are stored in different directories. 
- Docker now has an external volume for media and postgres data
- travis builds docker containers and published them on docker hub.
- Docker container is now running in production mode. Serving static assets via nginx. uwsgi is being held by supervisor.


#### How should this be manually tested?
Needs monkey testing. 

Need to deploy on dev machines and make sure that the changes don't break static files and media files.

#### What are the relevant tickets?
N/A

#### Screenshots (if appropriate)
#### Definition of Done:


#### Deploying on Existing Instances Notes
* Remove all the pyc files `find . -name *.pyc --delete` (this is actually a result from the API endpoint update).
* local_untracked.py
    * Make sure to add a SECRET_KEY to your local_untracked or env var (https://www.miniwebtool.com/django-secret-key-generator/)
    * If using FileSystemStorage, make sure to remove the `DEFAULT_FILE_STORAGE = 'django.core.files.storage.FileSystemStorage'` line from the local_untracked.py file.
* main.py
    * Change to using the `aws.py` settings file (instead of `main.py`) if you are using AWS resources. `main.py` was renamed `prod.py` and is now the bare minimum content for running the app production mode without AWS resources.

* run the deploy script
